src/Akka.Persistence.SqlServer/Journal/BatchingSqlServerJournal.cs

```diff
@@ -169,5 +184,43 @@ protected override SqlConnection CreateConnection(string connectionString)
         {
             return new SqlConnection(connectionString);
         }
+
+        /// <inheritdoc />
+        protected override void PreStart()
+        {
+            base.PreStart();
+
+            // if need to use constant parameters, prepare column sizes for parameter generation
+            if (_useConstantParameterSize)
+            {
+                using (var connection = CreateConnection(Setup.ConnectionString))
+                {
+                    _columnSizes = ColumnSizeLoader.LoadJournalColumnSizes(Setup.NamingConventions, connection);
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void CustomDbParameterSetup(SqlCommand command, DbParameter param)
+        {
+            if (!_columnSizes.HasValue)
+                return;
+            
+            // if column sizes are loaded, use them to define constant parameter size values
+            switch (param.ParameterName)
+            {
+                case "@PersistenceId":
+                    param.Size = _columnSizes.Value.PersistenceIdColumnSize;
+                    break;
+                
+                case "@Tag":
+                    param.Size = _columnSizes.Value.TagsColumnSize;
+                    break;
+                
+                case "@Manifest":
+                    param.Size = _columnSizes.Value.ManifestColumnSize;
```

Comment:
Here is where the magic happens: this parameter size is then used in queries and, when not set, is set dynamically based on `param.Value` size. Here we make it constant based on actual column size.